### PR TITLE
[WIP] sip_proxy: SIP protocol support in envoy

### PR DESCRIPTION
**Pull request by durd07**
For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/main/PULL_REQUESTS.md)

Commit Message:
[Issues]: https://github.com/envoyproxy/envoy/issues/16247

Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -71,6 +71,8 @@ extensions/filters/common/original_src @snowp @klarose
 # redis cluster extension
 /*/extensions/clusters/redis @msukalski @henryyyang @mattklein123
 /*/extensions/common/redis @msukalski @henryyyang @mattklein123
+# sip_proxy extension
+/*/extensions/filters/network/sip_proxy @durd07 @nearbyfly
```
[the remark this answers is not in the document]
Comment (antoniovicente) [reply CODEOWNERS:75]: Also, this extension needs tests.

[the remark this answers is not in the document]
Comment (durd07) [reply CODEOWNERS:75]: Hi @htuch @antoniovicente ,
thanks for your comments. this PR still in **Draft** stage, There are still many tasks to do.
I opened this PR to following the CI processes.

I will follow the procedure of https://github.com/envoyproxy/envoy/blob/main/EXTENSION_POLICY.md to do the change and add tests later.

Anyway, This PR is used to introduce SIP extension into Envoy, and this is the Design Issue Page: https://github.com/envoyproxy/envoy/issues/16247. 

@htuch , do you know how to identify a maintainer as the sponsor for the stratege review?

[the remark this answers is not in the document]
Comment (antoniovicente) [reply CODEOWNERS:75]: > Hi @htuch @antoniovicente ,
> thanks for your comments. this PR still in **Draft** stage, There are still many tasks to do.
> I opened this PR to following the CI processes.
> 
> I will follow the procedure of https://github.com/envoyproxy/envoy/blob/main/EXTENSION_POLICY.md to do the change and add tests later.

What part of the procedure are you referring to?  My read of the document above is that there's no defined process for accepting an extension without tests.

> Anyway, This PR is used to introduce SIP extension into Envoy, and this is the Design Issue Page: #16247.
> 
> @htuch , do you know how to identify a maintainer as the sponsor for the strategy review?
> 
> Currently, we can focus on the strategy review.

**Comment by htuch** [comment]
> @htuch , do you know how to identify a maintainer as the sponsor for the strategy review?

If you don't already know a maintainer who is likely to make sense, I suggest reaching out to envoy-maintainers@googlegroups.com.

**Comment by junr03** [comment]
@durd07 we really appreciate your contribution. In an effort to keep PRs actively engaged, I am closing PRs that don't have activity. Please re-open when you have worked out sponsorship (per @htuch), and want to continue this work stream.

**Pull request closed without merging**